src/main/groovy/life/qbic/variantstore/database/MariaDBVariantstoreStorage.groovy

```diff
@@ -2024,12 +2023,13 @@ gene.id = consequence_has_gene.gene_id INNER JOIN consequence on consequence_has
     private List<Gene> tryToStoreGeneObjects(List<Gene> genes) {
         Sql sql = requestNewConnection()
         sql.connection.autoCommit = false
-        sql.withBatch("insert INTO gene (symbol, name, biotype, chr, start, end, synonyms, geneid, description, strand, version) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE id=id") {
+        sql.withBatch("insert INTO gene (symbol, name, biotype, chr, start, end, synonyms, geneid, description, strand, version) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE symbol=?, name=?, " + "biotype=?, chr=?, start=?, end=?, synonyms=?, geneid=?, description=?, strand=?, version=?") {
             BatchingPreparedStatementWrapper ps ->
             genes.each { gene ->
                 ps.addBatch([gene.symbol, gene.name, gene.bioType, gene.chromosome, gene.geneStart,
-                             gene.geneEnd, gene.synonyms[0], gene.geneId, gene.description, gene
-                                     .strand, gene.version])
+                             gene.geneEnd, gene.synonyms[0], gene.geneId, gene.description, gene.strand, gene.version,
+                             gene.symbol, gene.name, gene.bioType, gene.chromosome, gene.geneStart,
+                             gene.geneEnd, gene.synonyms[0], gene.geneId, gene.description, gene.strand, gene.version])
             }
         }
 
```

Comment:
I think it would make sense to split up this class into multiple subclasses down the line dependent on the functionality(search, store etc.) 
Since it's over 2000 lines of code long it approaches the territory of a god class ;)

Reply:
I totally agree! In parallel I am working on an implementation for PostgreSQL which will use Micronaut Data and therefore will be (hopefully) much cleaner and need less code. When this works the plan is to use the same implementation for the MariaDB interface.